### PR TITLE
Implement api to get node status

### PR DESCRIPTION
**Pull request by Sindica**

### CODE REVIEW

File: resource-management/pkg/service-api/endpoints/constants.go [hunk 1]
```diff
@@ -22,9 +22,12 @@ const (
 
 	//RegionlessResourcePath is the default api service url
 	RegionlessResourcePath = "/resource"
-	ListWatchResourcePath  = RegionlessResourcePath + "/{clientid}"
-	UpdateResourcePath     = RegionlessResourcePath + "/{clientid}" + "/addResource"
-	ReduceResourcePath     = RegionlessResourcePath + "/{clientid}" + "/reduceResource"
+
+	NodeStatusPath = "/nodes?"
```
Comment (sonyafenge) [inline resource-management/pkg/service-api/endpoints/constants.go:26]: our initial design is using RegionlessResourcePath as root path, suggest to change to:
NodeStatusPath = RegionlessResourcePath +"/nodes?"

Comment (Sindica) [reply resource-management/pkg/service-api/endpoints/constants.go:26]: There are already two types of paths:
1. "/clients", used for client registration
2. "/resource", used for resources get/update

Querying for node status is a different type of request, it does not belong to any of the above.
We might need to add additional layer to nodes for large amount of node summary. The external APIs are designed to be provided from another service. This one is an internal API used for quick performance test.

Comment (sonyafenge) [reply resource-management/pkg/service-api/endpoints/constants.go:26]: Suggest to add TODO to finalize the url later

**Review by sonyafenge — APPROVED** [review]
/LGTM

File: resource-management/pkg/service-api/endpoints/constants.go (hunk 1 above)
Comment (Sindica) [reply resource-management/pkg/service-api/endpoints/constants.go:27]: > Suggest to add TODO to finalize the url later

Done.

File: resource-management/pkg/service-api/endpoints/installer.go [hunk 2]
```diff
@@ -123,6 +124,52 @@ func (i *Installer) handleClientUnRegistration(resp http.ResponseWriter, req *ht
...
+	klog.V(3).Infof("handle /resource/nodes/. URL path: %s", req.URL.Path)
+
+	switch req.Method {
+	case http.MethodGet:
+		regionName, rpName, nodeId := getNodeId(req)
+		resp.WriteHeader(http.StatusOK)
+		resp.Header().Set("Content-Type", "text/plain")
+
+		region := location.GetRegionFromRegionName(regionName)
+		resourceParition := location.GetPartitionFromPartitionName(rpName)
```
Comment (q131172019) [inline resource-management/pkg/service-api/endpoints/installer.go:137]: In my test environment, the line 137 return error:

ubuntu@ip-172-31-8-82:~/go/src/GRS$ /usr/local/go/bin/go run resource-management/cmds/service-api/service-api.go --master_ip=ec2-52-11-1-193.us-west-2.compute.amazonaws.com --resource_urls=ec2-54-187-22-27.us-west-2.compute.amazonaws.com:9119,ec2-35-91-52-72.us-west-2.compute.amazonaws.com:9119 --enable_metrics=false -v=3
# global-resource-service/resource-management/pkg/service-api/endpoints
resource-management/pkg/service-api/endpoints/installer.go:137:20: assignment mismatch: 1 variable but location.GetPartitionFromPartitionName returns 2 values

Comment (Sindica) [reply resource-management/pkg/service-api/endpoints/installer.go:137]: This error was caused independent implementation and auto code merge. Resolved in PR 148

**Pull request merged**